### DISCUSSION TITLE
Software dimming function resets to full brightness a few seconds when f.lux is running

### DISCUSSION THREAD

**Discussion by giulianofichera** [MonitorControl/MonitorControl#561] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

When enabling **software** dimming for an HDMI TV (connected through a USB C hub) the TV dims accordingly but suddenly returns to full brightness.
This bug occurs sooner when my mouse changes display, I click something or I move the mouse rapidly.
Even without performing any action, the TV returns to full brightness inevitably.
I have tried all possible settings combinations: changing the brightness using the keyboard, the software slider, changing the brightness for all screens and also independently.
Also tried a reboot, reinstall, reinstall + reboot, unplugging and plugging the usb hub.

### Steps to reproduce

1. Enable software dimming if required option.
2. Change brightness (using the keyboard shortcuts or the menu bar slider).
3. Move the mouse between screens to speed up the bug reproduction, or simply wait for it to happen some seconds after without any intervention.

### Expected behavior

The software brightness adjusts itself accordingly and remains in the desired brightness level until changed.

### Anything else?

https://user-images.githubusercontent.com/80286668/132044342-9ae1b5ff-ba66-4d05-b194-b4f1aa9a0bf2.mp4

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.5.2 Big Sur
- Mac model: MacBook Air (M1, 2020)
- MonitorControl version: 3.0.0 Build 3095
- Monitor(s): Philco Pld50us9a1 4k
- Apple Silicon/M1 (yes or no): yes

- USB C Hub used: Anker (https://www.amazon.com/Anker-PowerExpand-Adapter-Delivery-Ethernet/dp/B087QZVQJX/ref=sr_1_5?dchild=1&keywords=anker+usb+c+hub&qid=1630689207&sr=8-5)
```

**Comment by waydabber** [comment, marked as answer]
It seems like the gamma table is being reset for some reason immediately after changing it. The fact that it happens upon some activity is because on most macs gamma table changes are not reflected on the screen unless there is an event that causes some part of the screen to redraw. This is why you need to move the mouse or do something else that changes something on the screen and then the brightness is changed back to full.

Don't you use any other software that does gamma table manipulation? Maybe some screen management app or anything else that changes the displays, gamma, tint, etc of the display (like [f.lux](https://justgetflux.com))?

**Reply by giulianofichera** [reply]
Hi, Thanks for the quick response!
I am in fact using f.lux, completely forgot about it because it isn't active during daytime. I just tested again after quitting f.lux and the app behaves perfectly.

Is there any way to use both apps at the same time?
Once again, thank you very much for the detailed and quick answers!

**Comment by waydabber** [comment]
Hi @giulianofichera,

f.lux and MonitorControl can run side-by-side only if MonitorControl uses Hardware Control (DDC). F.lux is incompatible with Software Control and 'Further lower brightness via software dimming' as both apps use a technique called gamma table manipulation to achieve its goal and they cancel each other out. You can use MonitorControl with NightShift as it is doing mostly the same as f.lux but it is a system service which applies its effect over the manipulated gamma table so it behaves in a neutral way.

If you want to use f.lux and use software dimming, you might want to try [Quickshade](https://apps.apple.com/us/app/quickshade/id931571202?mt=12) which simply draws a black frame over the screen and does not affect gamma tables. The only drawback with Quickshade is that the result is not as convincing (but still fine) and the mouse cursor will remain bright (as it hovers over the black frame).

**Reply by giulianofichera** [reply]
Hi @waydabber !
I'm currently using Quickshade but it doesn't support dimming individual screens. Your solution is cleaner overall and would love to use it, I might ditch f.lux for it.
The issue with Night Shift is that I can't make it work on my TV, it only works for the built in display.

Thanks for the detailed explanation and for this amazing app!
